src/sawtoothMain/kotlin/lumberjack/sawtooth/component/PatternComponent.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * This file is part of Lumberjack.
+ *
+ * Lumberjack is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Lumberjack is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Lumberjack.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package lumberjack.sawtooth.component
+
+import lumberjack.sawtooth.event.LogEvent
+import lumberjack.sawtooth.layout.Layout
+
+public typealias PatternComponentInitialiser=(modifiers: List<String>) -> PatternComponent
```

Comment:
Instead of this, how about just a full on `PatternComponentFactory`?

```kotlin
interface PatternComponentFactory {

  // naming is whatever, i forget what log4j calls it
  fun withAlias(alias: String, modifiers: List<String>): PatternComponent?
}
```

Can then have a `DefaultPatternComponentFactory` with the ones you've defined in `PatternLayout`

```kotlin
object DefaultPatternComponentFactory : PatternComponentFactory {
  override withAlias(alias: String, modifiers: List<String>): PatternComponent? {
    return when (alias) {
      "m", "msg", "message" -> MessageComponent
      // etc
      else -> null
    }
  }
}
```

Then if users want to "add on" to the defaults

```kotlin
class CustomPatternComponent : PatternComponent {
  override fun withAlias(alias: String, modifiers: List<String>): PatternComponent? {
    // custom logic
    return customPattern ?: DefaultPatternComponentFactory.withAlias(alias, modifiers)
  }
}
```

Then just simply this one interface can be defined in the factory for `PatternLayout`.

Reply:
I guess Log4J calls them "conversion patterns" because they resemble C printf pattern formatting. I think that's a long winded name. I thin alias can work, what do you think?